procrustes/permutation.py

```diff
@@ -778,20 +769,20 @@ def permutation_2sided_explicit(array_a, array_b,
     used as a checker for small dataset.
 
     """
-    print("Warning: This brute-strength method is computational expensive! \n"
-          "But it can be used as a checker for a small dataset.")
+    print(
+        "Warning: This brute-strength method is computational expensive! \n"
+        "But it can be used as a checker for a small dataset."
+    )
```

Comment:
This could go to the note section in the function document and maybe "small matrices" would be more precise than small dataset.

Reply:
Good point. I think this is left-over from the case where the function was an error, and so it was nonnegative. We should probably go back to the user passing an optional number (which is an approximate lower bound on the function, and could be -Infinity) and the algorithm stops whenever the objective function goes below that bound.

Reply:
Good catch @Ali-Tehrani. It is now addressed in 3fa82e6.